rpc-rs/src/rpc_client.rs

```diff
@@ -279,14 +280,32 @@ impl RpcClient {
         // Record all of the fields on the span. To avoid extra allocations, we are only going to
         // record here after we generate/derive the values
         let span = tracing::span::Span::current();
-        span.record("provider_id", &tracing::field::display(&issuer));
+        if let Some(hb) = get_host_bridge_safe() {
+            span.record("provider_id", &tracing::field::display(hb.provider_key()));
+        }
         span.record("method", &tracing::field::display(&message.method));
         span.record("lattice_id", &tracing::field::display(&lattice));
-        span.record("target_id", &tracing::field::display(&target.public_key));
         span.record("subject", &tracing::field::display(&subject));
         span.record("issuer", &tracing::field::display(&issuer));
+        if !origin.public_key.is_empty() {
+            span.record("sender_key", &tracing::field::display(&origin.public_key));
+        }
+        if !target.contract_id.is_empty() {
+            span.record("contract_id", &tracing::field::display(&target.contract_id));
+        }
+        if !target.link_name.is_empty() {
+            span.record("link_name", &tracing::field::display(&target.link_name));
+        }
+        if !target.public_key.is_empty() {
+            span.record("target_key", &tracing::field::display(&target.public_key));
+        } else {
+            error!(op=%message.method, origin=?origin, "rpc send is missing target.public_key");
+            return Err(RpcError::Rpc(format!(
+                "rpc send is missing target public_key. op={}, origin={:?}",
+                &message.method, &origin
+            )));
+        }
```

Comment:
Each of the fields referenced in these `record` calls need to be initialized when the span is created (in this case, in the `instrument` macro). Unfortunately, fields cannot be added dynamically to a span. See the "Note" tucked away here: https://docs.rs/tracing/0.1.37/tracing/span/struct.Span.html#method.record